src/ol/expr/expression.js

```diff
@@ -613,6 +615,12 @@ const parsers = {
     parseArgsOfType(NumberType)
   ),
   [Ops.Palette]: createParser(ColorType, withArgsCount(2, 2), parsePaletteArgs),
+  [Ops.ToNumber]: createParser(
+    NumberType,
+    withArgsCount(1, 1),
+    parseArgsOfType(StringType | BooleanType),
+    parseToNumberArgs
+  ),
```

Comment:
This should be enough:

```suggestion
  [Ops.ToNumber]: createParser(
    NumberType,
    withArgsCount(1, Infinity),
    parseArgsOfType(StringType | BooleanType | NumberType)
  ),
```

Number arguments should be allowed since they can be useful in fallbacks.